R/checkMechanisms.R

```diff
@@ -0,0 +1,57 @@
+#' @export
+#' @title Check Mechanisms
+#'
+#' @description Tests mechanisms found in the Data Pack PSNUxIM tab
+#' against the Mechanism View from DATIM.
+#'
+#' @inheritParams datapackr_params
+#'
+#' @return d
+#'
+checkMechanisms <- function(d,
+                            cached_mechs_path = paste0(Sys.getenv("support_files_directory"), "mechs.rds"),
+                            d2_session = dynGet("d2_default_session",
+                                                inherits = TRUE)) {
+
+  mechs_data <- unique(d$datim$MER$attributeOptionCombo)
+
+  period_info <- datimvalidation::getPeriodFromISO(paste0(d$info$cop_year, "Oct"))
+
+  operating_unit <- getOperatingUnitFromCountryUIDs(d$info$country_uids)
+
+  mechs_datim <- datapackr::getMechanismView(d2_session = d2_session,
+                                             update_stale_cache = TRUE,
+                                             cached_mechs_path = cached_mechs_path) %>%
+    dplyr::filter(ou == operating_unit$ou) %>%
```

Comment:
See comment above. This would reference metadata in `d` object.
```suggestion
    dplyr::filter(ou == d$info$operating_unit$ou) %>%
```